### ISSUE TITLE
openni2 make java build optional

### ISSUE THREAD

**Issue by xzcvczx** [Homebrew/homebrew-science#4848]
Currently openni2 requires java for normal builds, This is only required for the java wrapper and java samples so As attached i have patched out the java stuff from being built if the option without-java is used.

2 patches are used. One for the brew recipe

```
diff --git a/openni2.rb b/openni2.rb
index e0f59d4..c2a68ef 100644
--- a/openni2.rb
+++ b/openni2.rb
@@ -7,13 +7,22 @@ class Openni2 < Formula
 
   option :universal
   option "with-docs", "Build documentation using javadoc (might fail with Java 1.8)"
+  option "without-java", "Build without the java wrappers and samples so that java is not required"
 
   depends_on :python
   depends_on "libusb"
   depends_on "doxygen" => :build if build.with? "docs"
+  depends_on :java => :recommended
 
   patch :DATA if build.without? "docs"
 
+  if build.without?(:java)
+    patch do
+      url "file:///Users/simon/Downloads/openni2-nojava.patch"
+      sha256 "ab7b36288651e91020fb441270b8a37e0efe191f4c7373e6634b63a6973aee6a"
+    end
+  end
+
   stable do
     patch do
       url "https://github.com/occipital/OpenNI2/pull/18.patch"
@@ -64,3 +73,4 @@ index 4ce9ed2..fad7017 100755
          
          # Include
          shutil.copytree(os.path.join(rootDir, 'Include'), os.path.join(self.outDir, 'Include'))
+
```

and openni2-nojava.patch being

```
diff --git a/Makefile.orig b/Makefile
index a436a84..dd7e41b 100644
--- a/Makefile.orig
+++ b/Makefile
@@ -54,7 +54,6 @@ ALL_CORE_PROJS = \
 	$(OPENNI) \
 	$(DEPTH_UTILS) \
 	$(ALL_DRIVERS) \
-	$(ALL_WRAPPERS) \
 	$(ALL_TOOLS)
 
 # list all samples
@@ -84,8 +83,7 @@ else
 endif
 
 ALL_SAMPLES = \
-	$(CORE_SAMPLES) \
-	$(JAVA_SAMPLES)
+	$(CORE_SAMPLES)
 
 # list all projects that are build
 ALL_BUILD_PROJS = \
diff --git a/Packaging/Harvest.py.orig b/Packaging/Harvest.py
index fad7017..9a38639 100755
--- a/Packaging/Harvest.py.orig
+++ b/Packaging/Harvest.py
@@ -82,8 +82,8 @@ class Harvest:
         os.makedirs(targetDir)
         # start with OpenNI itself
         self.copySharedObject(self.binDir, 'OpenNI2', targetDir)
-        self.copySharedObject(self.binDir, 'OpenNI2.jni', targetDir)
-        shutil.copy(os.path.join(self.binDir, 'org.openni.jar'), targetDir)
+        #self.copySharedObject(self.binDir, 'OpenNI2.jni', targetDir)
+        #shutil.copy(os.path.join(self.binDir, 'org.openni.jar'), targetDir)
         shutil.copy(os.path.join(self.rootDir, 'Config', 'OpenNI.ini'), targetDir)
         # and now all drivers
         binDriversDir = os.path.join(self.binDir, 'OpenNI2', 'Drivers')
@@ -294,7 +294,7 @@ $(OUTPUT_FILE): copy-redist
         self.copyGLUT(os.path.join(samplesDir, 'Bin'))
         self.copySample(samplesDir, 'SimpleRead')
         self.copySample(samplesDir, 'SimpleViewer', isGL = True)
-        self.copySample(samplesDir, 'SimpleViewer.java', isJava = True)
+        #self.copySample(samplesDir, 'SimpleViewer.java', isJava = True)
         self.copySample(samplesDir, 'EventBasedRead')
         self.copySample(samplesDir, 'MultiDepthViewer', isGL = True)
         self.copySample(samplesDir, 'MultipleStreamRead')

```

occipital/OpenNI2#104

**Comment by iMichka** [comment]
Thanks for your contribution.

I see that you opened a pull request here: #4863 

You did not include the patch there though. Any reason for this?

**Comment by xzcvczx** [comment]
This is adding an option, where as the pull request is just saying that java is required to build. As currently you just get the dialog saying how to install java

**Comment by iMichka** [comment]
Compilation fixes and update to 2.2.0-debian were done by #5192.

We will not support non-upstreamed patches. You will need to lobby with upstream to make java optional and to get your changes in a new, stable release. You could also host your own version of the formula as an alternative. Thanks for all the hard work on this.
